src/wasm-builder.h

```diff
@@ -699,10 +699,11 @@ class Builder {
   }
   RefNull* makeRefNull(HeapType type) {
     auto* ret = wasm.allocator.alloc<RefNull>();
-    ret->finalize(Type(type, Nullable));
+    ret->finalize(Type(type.getBottom(), Nullable));
     return ret;
   }
   RefNull* makeRefNull(Type type) {
+    assert(type.isNullable() && type.getHeapType().isBottom());
```

Comment:
This isn't quite the same, since `isNull` also includes non-nullable references to bottom types (which are not inhabitable). I'll change this to `type.isNullable() && type.isNull()`.

Reply:
Hmm, how can we have a non-nullable reference to a bottom type? (what does that look like in wasm?)

Reply:
`(ref none)` is a perfectly valid type, it's just not possible to create a value that inhabits it. So I could have a function that returns `(ref none)`, for instance, and know that it will never return.

Reply:
Oh, really.. I would have expected the wasm spec to disallow that. Just me I guess...